lib/gollum-lib/file.rb

```diff
@@ -19,13 +19,15 @@ def initialize(wiki)
     # Public: The url path required to reach this page within the repo.
     #
     # Returns the String url_path
+
     def url_path
-      path = self.path
-      if path.include?('/')
-        path = path.sub(/\/[^\/]+$/, '/')
-        path << name
-      end
-      path
+      path = if self.path.include?('/')
```

Comment:
I'm not a huge fan of this kind of assignment to be honest, since it requires a substantial look ahead to understand how the assignment is determined.

Reply:
Agreed